packages/repository-json-schema/src/build-schema.ts

```diff
@@ -14,23 +14,62 @@ import {
 import {JSONSchema6 as JSONSchema} from 'json-schema';
 import {JSON_SCHEMA_KEY} from './keys';
 
+export interface JsonSchemaOptions {
+  includeRelations?: boolean;
+  visited?: {[key: string]: JSONSchema};
```

Comment:
What's `visited` used for?

Reply:
In order to handle circular dependencies and don't overflow the stack, we need to know which models have been already visited.

```
1. get schema for ProductWithLinks
2. get schema for CategoryWithLinks (triggered by `ProductWithLinks#category`)
3. get schema for ProductWithLinks (triggered by `CategoryWithLinks#products`)
4. get schema for CategoryWithLinks (triggered by `ProductWithLinks#category`)
5. etc. until the stack overflows
```

With `options.visited`, the step 3 sees that `ProductWithLinks` has been already visited and emits the reference only.